src/frontend/src/components/common/Collapsible/index.tsx

```diff
@@ -11,15 +11,24 @@ const Collapsible = ({
   alignment,
   children,
   isVisible = false,
+  onVisibleChange,
 }: CollapsibleProps) => {
   const [isOpen, setIsOpen] = useState<boolean>(false);
-
   const handleToggle = (): void => {
     setIsOpen(!isOpen);
+
+    if (alignment === 'left') {
+      onVisibleChange(!isOpen, undefined);
+    }
+    if (alignment === 'right') {
+      onVisibleChange(undefined, !isOpen);
+    }
```

Comment:
Ideally, handling Left/rights is not the responsibility of Collapsable component.